### PR TITLE
[DOCS] OpenTelemetry - add OpenTelemetry protocol support content

### PR DESCRIPTION
**Pull request by EamonnTP**
### Summary

This PR updates the current OpenTelemetry integration [docs](https://www.elastic.co/guide/en/apm/get-started/current/open-telemetry-elastic.html) with the recent support for OTLP/gRPC on the standard port.

To accommodate the new content, I have restructured the content to include both the Elastic OpenTelemetry integrations: **Elastic exporter on the OpenTelemetry collector** and **APM Server native support of OpenTelemetry protocol**. I have also included updated architecture diagrams for both integrations. 

### Docs preview

https://apm-server_4964.docs-preview.app.elstc.co/guide/en/apm/get-started/master/open-telemetry-elastic.html

### Related issue

Closes https://github.com/elastic/apm-server/issues/4678

### CODE REVIEW

**Review by dedemorton — COMMENTED** [review]
I hope all my comments are applied properly. GitHub is showing duplicate comments. :-/

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 1]
```diff
@@ -176,33 +113,156 @@ This example configuration file accepts input from an OpenTelemetry Agent, proce
...
+     com.mycompany.checkout.CheckoutServiceServer
+----
+
+|=== 
+
+| `OTEL_RESOURCE_ATTRIBUTES` | The service name to identify your application.
+
+| `OTEL_EXPORTER_OTLP_ENDPOINT` | APM Server URL. The host and port that APM Server listens for events on.
+
+| `OTEL_EXPORTER_OTLP_HEADERS` | Authorization header that includes the Elastic APM Secret token or API key. Please note the required space between `Bearer` and `apm-secret-token`.
```
Comment (axw) [inline docs/guide/opentelemetry-elastic.asciidoc:218]: Perhaps provide an example of how to format an API Key too? And/or link to https://www.elastic.co/guide/en/apm/server/current/api-key.html

**Review by axw — COMMENTED** [review]
Looks good! I have only reviewed technical details.

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 1 above)
Comment (cyrille-leclerc) [reply docs/guide/opentelemetry-elastic.asciidoc:218]: Very good point. I thought about it and dropped the ball. 

Would it be like 
```
export OTEL_EXPORTER_OTLP_HEADERS="Authorization=ApiKey R0gzRWIzUUI3eVpiU054S3pYSy06bXQyQWl4TlZUeEcyUjd4cUZDS0NlUQ=="
```

Comment (axw) [reply docs/guide/opentelemetry-elastic.asciidoc:218]: @cyrille-leclerc yes that looks correct.

Re additional headers, I'm onboard with having method-specific headers. I'd still like to wait for feedback like we discussed recently.

Comment (cyrille-leclerc) [reply docs/guide/opentelemetry-elastic.asciidoc:218]: @EamonnTP can you please document the usage of API akei based authentication with:
``` 
export OTEL_EXPORTER_OTLP_HEADERS="Authorization=ApiKey R0gzRWIzUUI3eVpiU054S3pYSy06bXQyQWl4TlZUeEcyUjd4cUZDS0NlUQ=="
```

@axw are we sure that we can't simplify and also handle the api key authentication with the `OTEL_EXPORTER_OTLP_HEADERS="Authorization=Bearer xxx` format?

Comment (axw) [reply docs/guide/opentelemetry-elastic.asciidoc:218]: @cyrille-leclerc are you asking if we can support an additional auth header as well as Authorization? Yes we can. Maybe you could open an issue describing how you would like it to look, and we can discuss further there?

Comment (cyrille-leclerc) [reply docs/guide/opentelemetry-elastic.asciidoc:218]: I was not clear, my question is: can we use the same `Authorization=Bearer xxx` format for both Secret Token authentication and API Key authentication.
We would support `Authorization=Bearer xxx` with `xxx` being the "secret token" or the "api key"

Comment (axw) [reply docs/guide/opentelemetry-elastic.asciidoc:218]: Oh I see. `Authorization: ApiKey <api_key>` was chosen to be consistent with Elasticsearch. We could change that, but that's an even bigger discussion.

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 2]
```diff
@@ -14,76 +14,55 @@
...
-The `traces` and `metrics` pipelines define the path of trace data and metrics through your collector and bring all three of these components together.
+|=== 
 
-TIP: More information is available in the {ot-pipelines}[OpenTelemetry pipeline docs].
+| *Agents* | The OpenTelemetry agents instrument the applications and exports the telemetry data to the OpenTelemetry collector.
 
-A final note: `extensions` can also be enabled for tasks like monitoring the health of the collector.
-See the {ot-extension}[OpenTelemetry extension readme] for a list of supported extensions.
+| *OpenTelemetry collector* | Collects the telemetry data (https://opentelemetry.io/docs/collector/configuration/#a-namereceiversaimg-width35-srchttpsrawgithubcomopen-telemetryopentelemetryiomainiconography32x32receiverssvgimg-receivers[receivers])
+from the OpenTelemetry agent, defines optional transformations (https://opentelemetry.io/docs/collector/configuration/#a-nameprocessorsaimg-width35-srchttpsrawgithubcomopen-telemetryopentelemetryiomainiconography32x32processorssvgimg-processors[processors])
```
[the remark this answers is not in the document]
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:47]: Good point. It's a little confusing. I have rephrased the sentence. Hopefully it's a lot clearer.

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 3]
```diff
@@ -176,33 +113,156 @@ This example configuration file accepts input from an OpenTelemetry Agent, proce
...
+and <<open-telemetry-elastic-kibana,visualizing metrics>> in {kib}.
+
+[[open-telemetry-elastic-protocol]]
+==== APM Server native support of OpenTelemetry protocol
+
+experimental::[This feature is experimental and may be changed in a future release. It is only available in a self-managed environment.]
+
+The APM Server native support of OpenTelemetry protocol integration allows you to send collected telemetry data directly from your applications to APM Server.
+Trace data collected from your services and the metrics data collected from your applications and infrastructure are sent using the
+OpenTelemetry protocol.
```
[the remark this answers is not in the document]
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:187]: No this is different. This refers to the OpenTelemetry protocol (OTLP).

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 1 above)
Comment (cyrille-leclerc) [reply docs/guide/opentelemetry-elastic.asciidoc:218]: If it's a bigger discussion, let's park it for the moment :-) We will solve the challenge of clarity primarily with documentation (and hopefully with error messages)

Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc]: I updated this section. Hopefully it provides more clarity.

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/guide/opentelemetry-elastic.asciidoc:185]: "Elastic's protocol" refers to how Elastic APM Agents send data to a collector, like APM Server. "OTLP" refers to how OpenTelemetry agents send data to a collector, like the OTEL collector.

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 4]
```diff
@@ -14,76 +14,53 @@
...
 
 [float]
-[[open-telemetry-elastic-deployment-planning]]
-==== Plan your deployment
+[[open-telemetry-collector-config]]
+===== Download and configure the collector
 
-OpenTelemetry Collectors can be run as an Agent, or as standalone collectors.
-They can be deployed as often as necessary and scaled up or out.
+OpenTelemetry Collectors can be run an agents or as standalone collectors.
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:58]:
```suggestion
OpenTelemetry Collectors can be run as agents or as standalone collectors.
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 5]
```diff
@@ -176,33 +111,160 @@ This example configuration file accepts input from an OpenTelemetry Agent, proce
...
+| `insecure` | Disable verification of the server's identity if TLS is enabled. (optional)
+
+|===
+
+[float]
+[[instrument-apps-collector]]
+===== Instrument applications
+
+To export traces and metrics to the OpenTelemetry Collector, ensure that you have instrumented your services and applications
+with the OpenTelemetry API, SDK, or both. For example, if you are a Java developer, you need to execute your Java app using the
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:153]: What does execute mean in this context?

```suggestion
with the OpenTelemetry API, SDK, or both. For example, if you are a Java developer, you need to instrument your Java app using the
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 6]
```diff
@@ -176,33 +111,160 @@ This example configuration file accepts input from an OpenTelemetry Agent, proce
...
+|=== 
+
+| `OTEL_RESOURCE_ATTRIBUTES` | The service name to identify your application.
+
+| `OTEL_EXPORTER_OTLP_ENDPOINT` | APM Server URL. The host and port that APM Server listens for events on.
+
+|===
+
+You are now ready to collect <<open-telemetry-elastic-traces-metrics,traces and metrics>>, <<open-telemetry-elastic-verify,verifying metrics>>,
+and <<open-telemetry-elastic-kibana,visualizing metrics>> in {kib}.
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:176]: Maybe remove these gerunds.

```suggestion
You are now ready to collect <<open-telemetry-elastic-traces-metrics,traces and metrics>>, <<open-telemetry-elastic-verify,verify metrics>>,
and <<open-telemetry-elastic-kibana,visualize metrics>> in {kib}.
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 7]
```diff
@@ -176,33 +111,160 @@ This example configuration file accepts input from an OpenTelemetry Agent, proce
...
 
-NOTE: For more information about getting started with an OpenTelemetry Collector,
-see the {ot-collector}[OpenTelemetry collector] docs.
+[[open-telemetry-elastic-protocol]]
+==== APM Server native support of OpenTelemetry protocol
+
+experimental::[This feature is experimental and may be changed in a future release. It is only available in a self-managed environment.]
+
+The APM Server native support of OpenTelemetry protocol integration allows you to send collected telemetry data directly from your applications to APM Server.
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:183]: Maybe one of these instead?

> The APM Server native support of the OpenTelemetry protocol allows you to send collected telemetry data directly from your applications to APM Server.

> The APM Server native OpenTelemetry protocol integration allows you to send collected telemetry data directly from your applications to APM Server.

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 8]
```diff
@@ -176,33 +111,160 @@ This example configuration file accepts input from an OpenTelemetry Agent, proce
...
+image::images/open-telemetry-protocol-arch.png[OpenTelemetry Elastic protocol architecture diagram]
 
 [float]
-[[open-telemetry-elastic-caveats]]
-==== Caveats
+[[instrument-apps-apm-server]]
+===== Instrument applications
+
+To export traces and metrics to APM Server, ensure that you have instrumented your services and applications
+with the OpenTelemetry API, SDK, or both. For example, if you are a Java developer, you need to execute your Java app using the
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:194]: `execute` again. It's possible this word is correct and I just haven't heard it yet.

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 9]
```diff
@@ -176,33 +111,160 @@ This example configuration file accepts input from an OpenTelemetry Agent, proce
...
+For information on how to format an API key, see our {apm-server-ref-v}/api-key.html[API key] docs.
+
+Please note the required space between `Bearer` and `apm_secret_token`, and `APIKey` and `api_key`.
+
+| `OTEL_EXPORTER_OTLP_CERTIFICATE` | Certificate for TLS credentials of the gRPC client. (optional)
+
+|===
+
+You are now ready to collect <<open-telemetry-elastic-traces-metrics,traces and metrics>>, <<open-telemetry-elastic-verify,verifying metrics>>,
+and <<open-telemetry-elastic-kibana,visualizing metrics>> in {kib}.
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:227]: Same gerunds as before. Sounds a little off.

**Review by bmorelli25 — COMMENTED** [review]
Looking great! A few suggestions.

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 10]
```diff
@@ -176,33 +111,160 @@ This example configuration file accepts input from an OpenTelemetry Agent, proce
...
+To export traces and metrics to the OpenTelemetry Collector, ensure that you have instrumented your services and applications
+with the OpenTelemetry API, SDK, or both. For example, if you are a Java developer, you need to execute your Java app using the
+https://github.com/open-telemetry/opentelemetry-java-instrumentation[OpenTelemetry agent for Java]. 
+
+By defining the following environment variables, you can customize the OTLP endpoint the agent will use to communicate with
+APM Server.
+
+[source,bash]
+----
+export OTEL_RESOURCE_ATTRIBUTES=service.name=frontend,service.version=1.0-SNAPSHOT,deployment.environment=staging
```
Comment (cyrille-leclerc) [inline docs/guide/opentelemetry-elastic.asciidoc:161]: `service.version=1.0-SNAPSHOT` -> `service.version=1.1.0`

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 7 above)
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:183]: Good suggestions. I went with the first one.

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 8 above)
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:194]: `Instrument` is a much better word here.

**Review by cyrille-leclerc — APPROVED** [review]

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 11]
```diff
@@ -14,76 +14,53 @@
...
-The `traces` and `metrics` pipelines define the path of trace data and metrics through your collector and bring all three of these components together.
+|=== 
 
-TIP: More information is available in the {ot-pipelines}[OpenTelemetry pipeline docs].
+| *Agents* | The OpenTelemetry agents instrument the applications and export the telemetry data to the OpenTelemetry collector.
 
-A final note: `extensions` can also be enabled for tasks like monitoring the health of the collector.
-See the {ot-extension}[OpenTelemetry extension readme] for a list of supported extensions.
+| *OpenTelemetry collector* | The https://opentelemetry.io/docs/collector/configuration/#a-namereceiversaimg-width35-srchttpsrawgithubcomopen-telemetryopentelemetryiomainiconography32x32receiverssvgimg-receivers[receiver]
+collects the telemetry data from the OpenTelemetry agent, and then the https://opentelemetry.io/docs/collector/configuration/#a-nameprocessorsaimg-width35-srchttpsrawgithubcomopen-telemetryopentelemetryiomainiconography32x32processorssvgimg-processors[processor]
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:46]: Fixing anchors

```suggestion
| *OpenTelemetry collector* | The https://opentelemetry.io/docs/collector/configuration/receivers[receiver]
collects the telemetry data from the OpenTelemetry agent, and then the https://opentelemetry.io/docs/collector/configuration/#processors[processor]
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 12]
```diff
@@ -14,76 +14,53 @@
 // Make tab-widgets work
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:14]: The `:ot-pipelines:` link on line 7 needs to be updated to `https://opentelemetry.io/docs/collector/configuration/#service`

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 11 above)
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc]: Great. Thanks!

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 12 above)
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:14]: Updated.

**Review by bmorelli25 — APPROVED** [review]

**Pull request merged**